scipy/cluster/tests/test_hierarchy.py

```diff
@@ -322,79 +322,89 @@ def test_leaders_single(self, xp):
 
 class TestIsIsomorphic:
 
-    @skip_if_array_api
-    def test_is_isomorphic_1(self):
+    @skip_if_xp_not_numpy(reason='`is_isomorphic` only supports NumPy backend')
+    @array_api_compatible
+    def test_is_isomorphic_1(self, xp):
         # Tests is_isomorphic on test case #1 (one flat cluster, different labellings)
-        a = [1, 1, 1]
-        b = [2, 2, 2]
-        assert_(is_isomorphic(a, b))
-        assert_(is_isomorphic(b, a))
+        a = xp.asarray([1, 1, 1])
+        b = xp.asarray([2, 2, 2])
+        assert is_isomorphic(a, b)
+        assert is_isomorphic(b, a)
 
-    @skip_if_array_api
-    def test_is_isomorphic_2(self):
+    @skip_if_xp_not_numpy(reason='`is_isomorphic` only supports NumPy backend')
```

Comment:
I guess we lose some mapping granularity with all the `pytest` abstraction: 

`SKIPPED [27] scipy/conftest.py:159: is_isomorphic only supports NumPy backend`

(i.e., the skip doesn't map to the module where it happened)

but that seems like something for a follow-up if it matters

Reply:
The granularity is maintained with `--verbose`, no?

Reply:
I didn't check that, but with `-rsx` the in-house skips are different than built-in ones I think? In any case, I believe that was the case before this PR as well.

Reply:
Oh, yeah `--verbose` would probably work, but the flags that produce just a short summary at the end are definitely different vs. "builtin" skips I think.